pkg/serializer/func_queue.go

```diff
@@ -81,6 +85,17 @@ func (fq *FunctionQueue) Stop() {
 	close(fq.stopCh)
 }
 
+// Wait until the FunctionQueue is stopped, or the specified context deadline
+// expires. Returns the error from the context, or nil if the FunctionQueue
+// was completed before the context deadline.
+func (fq *FunctionQueue) Wait(ctx context.Context) error {
+	select {
+	case <-fq.stopCh:
+	case <-ctx.Done():
+	}
+	return ctx.Err()
```

Comment:
I'm assuming that this will return a generic `context deadline exceeded`? It would be helpful to return a more specific error so we know where it is coming from

Reply:
I think this is up to the caller. If the context isn't exceeded, then this returns nil, which is the "Done" case.

Reply:
Changed my mind, new patch on top.